corehq/apps/api/resources/v0_5.py

```diff
@@ -522,3 +526,71 @@ def obj_get(self, bundle, **kwargs):
     class Meta(CustomResourceMeta):
         list_allowed_methods = []
         detail_allowed_methods = ["get"]
+
+
+UserDomain = namedtuple('UserDomain', 'domain_name project_name')
+UserDomain.__new__.__defaults__ = ('', '')
+
+
+class UserDomainsResource(Resource):
+    domain_name = fields.CharField(attribute='domain_name')
+    project_name = fields.CharField(attribute='project_name')
+
+    class Meta:
+        resource_name = 'user_domains'
+        authentication = ApiKeyAuthentication()
+        object_class = UserDomain
+        include_resource_uri = False
+
+    def obj_get_list(self, bundle, **kwargs):
+        return self.get_object_list(bundle.request)
+
+    def get_object_list(self, request):
+        couch_user = CouchUser.from_django_user(request.user)
+        results = []
+        for domain in couch_user.get_domains():
+            if not ZAPIER_INTEGRATION.enabled(domain):
+                continue
+            domain_object = Domain.get_by_name(domain)
+            results.append(UserDomain(
+                domain_name=domain_object.name,
+                project_name=domain_object.hr_name or domain_object.name
+            ))
+        return results
+
+
+Form = namedtuple('Form', 'form_xmlns form_name')
+Form.__new__.__defaults__ = ('', '')
+
+
+class DomainForms(Resource):
+    form_xmlns = fields.CharField(attribute='form_xmlns')
+    form_name = fields.CharField(attribute='form_name')
+
+    class Meta:
+        resource_name = 'domain_forms'
+        authentication = ApiKeyAuthentication()
+        object_class = Form
+        include_resource_uri = False
+
+    def obj_get_list(self, bundle, **kwargs):
+        application_id = bundle.request.GET.get('application_id')
+        if not application_id:
+            return []
```

Comment:
Should this not raise a 404 or some other error to indicate that app ID is required?